### ISSUE TITLE
Load Config Error

### ISSUE THREAD

**Issue by Mayo99** [coder-alpha/FMoviesPlus.bundle#27]
Hey Code-Alpha, On MAC, with FMoviesPlus v.0.64 I seem to have an issue doing either a "Load Bookmarks" or "Load Config" of either immediately after a save, and validated file was saved to ~/Library/Application Support/Plex Media Server/Plug-ins/FMoviesPlus.bundle/Contents/Resources/ . 

I need to Edit then Load Bookmarks, in order to get ride of a bad Bookmark that gives me an "Error: The page was not received. Please try again."

Attached is last 100 lines of com.plexapp.plugins.fmoviesplus.log:

[com.plexapp.plugins.fmoviesplus.log_LoadConfig_Error.log](https://github.com/coder-alpha/FMoviesPlus.bundle/files/1913131/com.plexapp.plugins.fmoviesplus.log_LoadConfig_Error.log)

Thanks in advance for looking it over, and resolving,

**Comment by coder-alpha** [comment]
Not finding the error you mention in the logs. There is a read error it seems. Can you try it again with debug enabled from the channel prefs and try to replicate the issue.

**Comment by Mayo99** [comment]
I see the same lines with debug enabled:

_**2018-04-16 13:03:19,315 (10bc13000) :  
CRITICAL (tools:518) - tools.py>LoadConfig >> :  >>> bad argument type for built-in operation (most recent call last):
  File "/Users/admin/Library/Application Support/Plex Media Server/Plug-ins/FMoviesPlus.bundle/Contents/Code/tools.py", line 422, in LoadConfig
    with io.open(bkup_file, 'r') as f:
TypeError: bad argument type for built-in operation**_

I already tried updating permissions to /FMoviesPlus.bundle/Contents/Resources/*.json 
with chmod 777 *.json

-rwxrwxrwx  1 admin  staff   30338 Apr 15 10:33 bookmarks.json
-rwxrwxrwx  1 admin  staff  378934 Apr 15 10:58 config.json

**Comment by Mayo99** [comment]
and while doing Load Bookmarks (with Enable Debug Mode) log shows:

**_2018-04-16 13:09:05,876 (10bc13000) :  CRITICAL (runtime:889) - Exception (most recent call last):
  File "/Applications/Plex Media Server.app/Contents/Resources/Plug-ins-1046ba85f/Framework.bundle/Contents/Resources/Versions/2/Python/Framework/components/runtime.py", line 843, in handle_request
    result = f(**d)
  File "/Users/admin/Library/Application Support/Plex Media Server/Plug-ins/FMoviesPlus.bundle/Contents/Code/tools.py", line 82, in DevToolsC
    Log(u'\n----------Loaded Config from {}----------'.format(resources_path))
  File "/Users/admin/Library/Application Support/Plex Media Server/Plug-ins/FMoviesPlus.bundle/Contents/Code/tools.py", line 238, in LoadBookmarks
    with io.open(bkup_file, 'r') as f:
TypeError: bad argument type for built-in operation_**

Again, the two .json files that were "saved" have been give Full Permissions.

**Comment by coder-alpha** [comment]
I'm able to save/load fine on my Win7 machine, can you try `chown plex:plex -R FMoviesPlus.bundle` and then try the same.

**Comment by Mayo99** [comment]
“ plex ” is not a user or group on my Mac, so `chown -R plex:plex FMoviesPlus.bundle` throws error `
chown: plex: illegal group name`.   My permissions of “ admin:staff “ are basically the same as “root”, and have full access to all files on the system,  PMS runs under Administrator login.

I run a Mid 2011 Mac mini Server (OS X Mavericks) v10.9.5 with OS X Server v3.2.2.  Plex Media Server is 64 Bit v1.12.1.4885, and I’ve been using Plex for years.  Usually I’m able to troubleshoot my own issues and/or provide feedback to Plex and/or Channel Developers.

I’m providing a fresh “com.plexapp.plugins.fmoviesplus.log” taken after a PMS restart, however, before the restart I updated Options with  ”Enable Debug Mode”, and even removed “Auth Admin through Plex.tv” for this test. Then after PMS & FMoviesPlus restart I clicked on “Options > Tools > Save Config” validated Config was saved, then did “Options > Tools > Load Config” and see the same issue show in the log.  I then updated Options back to “Auth Admin through Plex.tv” and removed “Debug Mode”, did another “Load Config” and the screen shoed the same “error: Could not Load Config file (config.json)” 

[com.plexapp.plugins.fmoviesplus.log.zip](https://github.com/coder-alpha/FMoviesPlus.bundle/files/1917750/com.plexapp.plugins.fmoviesplus.log.zip)

I think you once said you have a OS X VM, possibly you can use it to determine this issue.  If you want I can ask in  FMovies Plus Plex Forum to see if anyone running a MAC can attempt to duplicate issue, however, sometimes forum users don’t understand techie terms.

Also, if there’s another way other than an edit of ”config.json” or “bookmarks.json” save file, and load to remove a bad Bookmark please let me know, however, I’ve tried to find where bookmarks are actually saved in the .bundle and can’t find.

Thanks,

**Comment by coder-alpha** [comment]
OK, I'll test it on my VM, just give me a couple of days. From reading it seem to be an issue with the file encoding on OSX.

**Comment by coder-alpha** [comment]
I'm attaching a file `tools.py` (please rename it back from `.txt`) that you'd need to replace under `..\FMoviesPlus.bundle\Contents\Code\` that should fix the issue. I will have it included in the next release. 

Tested on OSX 10.12.6. Kindly confirm. Thanks.

[tools.py.txt](https://github.com/coder-alpha/FMoviesPlus.bundle/files/1933062/tools.py.txt)

**Comment by Mayo99** [comment]
WORKS!!!!  In case you wish to look at my log here it is:

[com.plexapp.plugins.fmoviesplus.log.zip](https://github.com/coder-alpha/FMoviesPlus.bundle/files/1933237/com.plexapp.plugins.fmoviesplus.log.zip)

BTW- I did the "edit" of bookmarks.json I wanted to after I did a save, then I did a "Load Bookmarks" again, other than seeing all my bookmarks a few times because of multiple "Load" done on Config & Bookmarks seen in the log, I did a "Clear Bookmarks" in Bookmarks and ran "Load Bookmarks" one again.  I wasn't expecting the "Load" of either Bookmarks or Config to duplicate entries, but not. big deal.  Was anything else duplicated by my "Load Confg" that I should be aware of?
 
In case you ever need a MAC tester feel free to reach out.

And Thanks for the Support, I'm a former SS-Plex user, and since Mike hasn't put effort into updating I'm a 100% FMoviesPlus user now.

**Comment by Mayo99** [comment]
oops, spoke to soon...  34 out of my original 40 My Bookmarks show"`Error: The page was not received. Please try again.`" 

Providing .log (after PMS restart), and both my bookmarks.json and config.json in attached .zip (".json" save(s) were done after applying the updated tool.py file).

[Archive.zip](https://github.com/coder-alpha/FMoviesPlus.bundle/files/1933906/Archive.zip)

I'll await your reply before I clear all Bookmarks, and add them back in manually.

**Comment by Mayo99** [comment]
So I decided easier today that since I’m now able to “Load” either "Config” or “Bookmarks” with updated `tools.py`, and since I was having other strange off and on issues, that I did a uninstall, reinstall of FMoviesPlus using WebTools.  After the fresh install I confirmed I wasn’t having the strange bookmark issues by added a few, deleting a few, created a few, and created more form from “Other Seasons” from within the same TV shows Title bookmark. Once I validated, I then shutdown PMS and updated `/FMoviesPlus.bundle/Contents/code` with the `tools.py` file you provided.

After PMS restated I did a “Load Config”, and all was good with my Bookmarks this time.  I then did some file editing of a `bookmarks.json` I saved off last week, deleting out lines of shows that gave me issues before.  Then I cleared all Bookmarks and did a “Load Bookmarks”, then after finding shows I edited out and adding them back as a Bookmark, everything seems to be working fine.

The issue I was originally having, and had with my old install after doing a Load Bookmarks using your updated tools.py code was very strange.  I was seeing “Add Bookmark” in a lot of current bookmarks already in Bookmarks.  When I clicked “Add” I then had 2 bookmarks, and one only add both showed “Remove Bookmark”, however, after doing remove to both, one was back in Bookmarks with “Add BookmarK.  I was also getting "`Error: The page was not received. Please try again.`”  on too many Bookmarks too.  So when I was reviewing logs, most of the bookmarks I had issues with before, and after my initial import/load with your updated tool.pf code that showed with an errors seemed to be an extra www in the domain name for ex. https://www1.www4.fmovies.se/search. or https://www4.www1.fmovies.se/search.

**Bottom line:**
1. Updated “tools.py” code works on OS X in doing a Load Config, or Load Bookmarks
2. Issues resolved with Bookmarks by doing a Save Config, save off `config.json file within `/Contents/Resources`, then doing an Uninstall/Reinstall moving back “Load Config” into `/Contents/Resources` and doing a “Load Config”
3. If issues continue then “Save Bookmarks”, make a copy and edit the “`/Contents/Resources.bookmarks.json`” file with a text editor and remove from “` { `“ to “` } `” rows that contained the bookmark that had an issue.  After saving “Bookmarks > Clear Bookmarks” then “Options > Tools > Load Bookmarks”.

**Comment by coder-alpha** [comment]
I have posted a [dev release](https://forums.plex.tv/discussion/comment/1659294/#Comment_1659294) incase you'd like to help test it out.
